backend-lucene/src/main/java/org/hibernate/search/v6poc/backend/lucene/search/sort/impl/SearchSortFactoryImpl.java

```diff
@@ -64,6 +66,20 @@ public SearchSort toSearchSort(SearchSortContributor<LuceneSearchSortCollector>
 		);
 	}
 
+	@Override
+	public DistanceSortBuilder<LuceneSearchSortCollector> distance(String absoluteFieldPath, GeoPoint location) {
+		LuceneIndexSchemaFieldNode<?> schemaNode = searchTargetModel.getSchemaNode( absoluteFieldPath );
+		if ( !schemaNode.getSortContributor().supportsSortingByDistance() ) {
+			throw log.distanceOperationsNotSupportedByFieldType( absoluteFieldPath );
```

Comment:
I don't really like the idea of introducing a spatial-specific getter just for that, but more importantly I don't see the point. For field sorts on geo point fields, we already throw an error late (when calling `contribute` on the builder), so why wouldn't we do the same for distance sorts on non-geopoint fields, and throw an error when contributeDistanceSort is called?

Alternatively, if you think early failure has value in this case, we could introduce `SearchSortBuilderFactories` like we do for predicates. In this case that would only require two factories:

* a "simple"/"standard" one with a "field sort" factory method that returns a builder and a "distance sort" factory method that throws an error
* a "geopoint" one with a "field sort" factory method that throws an error and a "distance sort" factory method that returns a builder

I think that would be more future-proof regardless of whether we want early errors or not, by the way, but that's just my opinion.